src/frontend/js/utils/api/lms/openedx-hawthorn.spec.ts

```diff
@@ -156,6 +157,34 @@ describe('OpenEdX Hawthorn API', () => {
           new Error('[SET - Enrollment] > 500 - Internal Server Error'),
         );
       });
+
+      it('throws localized HttpError if user is unauthorized to enroll', async () => {
+        fetchMock.post(`${EDX_ENDPOINT}/api/enrollment/v1/enrollment`, {
+          status: 400,
+          body: { localizedMessage: 'You are not authorized to enroll in this course' },
+        });
+
+        await expect(
+          HawthornApi.enrollment.set(`https://demo.endpoint/courses?course_id=${courseId}`, {
+            username,
+          }),
+        ).rejects.toThrow(
+          new HttpError(400, 'Bad Request', 'You are not authorized to enroll in this course'),
+        );
+      });
+
+      it('throws HttpError with no localized payload message if user is unauthorized to enroll', async () => {
+        fetchMock.post(`${EDX_ENDPOINT}/api/enrollment/v1/enrollment`, {
+          status: 400,
+          body: { message: 'Bad Request' },
+        });
+
+        await expect(
+          HawthornApi.enrollment.set(`https://demo.endpoint/courses?course_id=${courseId}`, {
+            username,
+          }),
+        ).rejects.toThrow(new HttpError(400, 'Bad Request'));
+      });
```

Comment:
Can you write a test to check what happend when the response returned has a 400 status code but the body is empty or something else than a json.